### DISCUSSION TITLE
svg

### DISCUSSION THREAD

**Discussion by eylles** [eylles/rxiv#10] [Ideas]
@Dhruv-Vanjari would you lend me a hand here, what can we do to make svg support a bit better? :D

**Comment by Dhruv-Vanjari** [comment]
Sure,

sxiv uses the `img->h`, `img->w`, and `img->zoom` properties for performing operations like zoom, pan, etc.
But in svg, we have the actual image size, and the size of the scaled raster image in `svg_t`.
This is the main reason we are having all these conflicts between different operations.
I have modified the `img_render` function to properly render svg documents, but others like fit_win, zoom, still need some work.

You should start by understanding how the calculations behind these operations work, so you can modify them to work with svg.

I was thinking about having a variable for the render dimentions, and use those for svg as well as raster images, so none would cause issues.
:)

**Reply by eylles** [reply]
just to see if i'm understanding things correctly, the variable for render dimensions would depend from the source to destination offset calculated in `img_render`? i honestly gave a quick look to the code for svg support, tought sounds nice and merged it without much after tought :D

**Reply by Dhruv-Vanjari** [reply]
Well the render dimension is `image size * zoom`. sxiv currently calculates the render dimensions wherever they are needed. What we need to do is calculate and store the render dimensions when the images is loaded or zoomed. This means that they need to be calculated in the image loaders and in the zoom function. And then use the render dimension variable wherever needed. This will eliminate the need of the `zoom` variable inside the `operation functions` like render, pan, fit_win, etc which is different for svg (`1.0`) and raster (`img->zoom`).

If you have any ideas how to go about this, I would love to hear them :)

**Reply by eylles** [reply]
hmmm, one of my objectives with the rxiv fork is to embed lua for configuration, the other is to have the freedom to break away from the sxiv way when needed, this seems like a necesary break away to smooth things out and join the calculations to one variable instead of having to calculate separately for svg and raster, dunno but it seems you have things tought out and are rather waiting for a green flag, in that case i say go ahead.
